src/util/metrics.ts

```diff
@@ -0,0 +1,60 @@
+import express from 'express';
+import { Application, Request, Response } from 'express';
+import client from 'prom-client';
+
+import { Log } from '../logging/Log';
+import { parseArgs } from '../parseArgs';
+
+export const httpErrorCounter = new client.Counter({
+	name: 'sas_http_errors',
+	help: 'Number of HTTP Errors',
+});
+
+interface IAppConfiguration {
+	port: number;
+	host: string;
+}
+
+export default class Metrics_App {
+	private app: Application;
+	private readonly port: number;
+	private readonly host: string;
+
+	/**
+	 * @param appConfig configuration for app.
+	 */
+	constructor({ host }: IAppConfiguration) {
+		const args = parseArgs();
+
+		this.port = Number(args.prometheus_port);
+		this.app = express();
+		this.host = host;
+
+		this.metricsEndpoint();
+	}
+
+	listen(): void {
+		const { logger } = Log;
```

Comment:
This is fine with the Logger since it's only being used in here which makes sense. But that being said as a side note, the logger can be a top level variable since it's only ever instantiated once. The logger file itself has a static `this.create` which only gets called the first time the logger is instantiated in the entry file.